numpyro/contrib/hsgp/util.py

```diff
@@ -0,0 +1,10 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Union
+
+import numpy as np
+
+import jax
+
+ARRAY_TYPE = Union[jax.Array, np.ndarray]  # jax.Array covers tracers
```

Comment:
I wonder if this type can be used in other NumPyro modules.

Reply:
Using `ArrayImpl` is only an issue when a model gets compiled and the arrays turn into tracers. `isinstance(X, jax.Array)` will work for both jax arrays and tracers.

Reply:
Some details here https://jax.readthedocs.io/en/latest/jax_array_migration.html

I believe this is best practice for typing jax arrays, but I am not sure

Reply:
I am definitively not an expert in type hints, so following the recommendation from the docs seems the safest path :)

Reply:
works for me!